Remove :_type_disabled for models in migrations, where type column doesn't exist

lpichler opened the pull request:
related https://github.com/ManageIQ/manageiq/issues/6739
based on comments:
https://github.com/ManageIQ/manageiq/pull/5554#discussion_r45544458 or https://github.com/ManageIQ/manageiq/issues/6739#issuecomment-185972452

this script helpded me:
```ruby
# prepare
#system("mkdir db/migrate_temp")
#Dir.glob("db/migrate/*").each do |f|
#  next if File.directory?(f)
#  next if f.include?("20130923182042_collapsed_initial_migration.rb")
#  f_with_new_path = f.gsub('migrate', 'migrate_temp')
#  system("mv #{f} #{f_with_new_path}")
#end
# system("rake db:migrate")

Dir.glob("db/migrate_temp/*").each do |f|
  next if File.directory?(f)
  f_with_new_path = f.gsub('migrate_temp', 'migrate')
  system("cp #{f} #{f_with_new_path}")
  system("rake db:migrate &>/dev/null")
  migration_info = get_migration_info(f_with_new_path)
  if migration_info.present?
    migration_info.map! do |klass_stub|
      klass_stub[:suggestion] = klass_stub[:sti_should_be_disabled] ? "ADD it -> " : "REMOVE it -> "
      klass_stub[:where] = "#{klass_stub[:f]} for #{klass_stub[:klass_stub]}"
      puts klass_stub[:suggestion] + klass_stub[:where]
    end
  end
end

def get_migration_info(f)
  type_column_exists = lambda { |t| ActiveRecord::Base.connection.column_exists?(t, :type) }
  table_exist = lambda { |t| ActiveRecord::Base.connection.data_source_exists?(t) }
  sti_disabled_in_stub = lambda do |content|
    content.include?("self.inheritance_column = :_type_disabled") ||
        content.include?("MigrationStubHelper")
  end
  matches = File.read(f).scan(/\s+class (.*) < ActiveRecord::Base([\s\S]*?)end/)
  if matches.present?
    out_array = []
    matches.each do |match|
      klass_stub, klass_content = match
      tbl_exist                 = table_exist.call(klass_stub.tableize.to_sym)
      sti_dis_in_stub           = sti_disabled_in_stub.call(klass_content)
      sti_should_be_disabled    = tbl_exist ? type_column_exists.call(klass_stub.tableize.to_sym) : false

      unless sti_should_be_disabled == sti_dis_in_stub
        out_array.push({:f => f, :klass_stub    => klass_stub,
                       :table_exist            => tbl_exist,
                       :sti_disabled_stub      => sti_dis_in_stub,
                       :sti_should_be_disabled => sti_should_be_disabled})
      end
    end
    out_array
  else
    []
  end
end
```

cc @Fryguy @jrafanie

lpichler commented:
@miq-bot add_label wip

lpichler commented:
@miq-bot remove_label wip

Fryguy commented on db/migrate/20150224192816_migrate_provisioning_manager_to_ems.rb:
```diff
@@ -8,9 +8,7 @@ class ProvisioningManager < ActiveRecord::Base
-  class OperatingSystemFlavor < ActiveRecord::Base
-    self.inheritance_column = :_type_disabled
-  end
+  class OperatingSystemFlavor < ActiveRecord::Base; end
```
Really surprised on this one :)

Fryguy commented:
@lpichler So does your script there check based on the latest tables, or what the state of the tables are *at that time*.

lpichler commented:
@Fryguy no but now yes, I updated it in way that it based on state of tables at that time.

I also removed `self.inheritance_column = :_type_disabled`  from `MigrationStubHelper` because only ExtManagementSystem(20131107000917_expand_dialog_field_default_value_size.rb) and DialogField(
20150407144345_add_kerberos_to_ext_management_system.rb) have needed it

jrafanie commented on db/migrate/20140409134713_move_log_collection_depot_settings_to_file_depot.rb:
```diff
@@ -5,20 +5,14 @@ class Authentication < ActiveRecord::Base
   end
 
-  class FileDepot < ActiveRecord::Base
-    self.inheritance_column = :_type_disabled # disable STI
```
We might want to keep this for now...

The migration on the following day adds a type column:
```
db/migrate/20140410132430_subclass_file_depot_by_protocol.rb-  def up
db/migrate/20140410132430_subclass_file_depot_by_protocol.rb:    add_column :file_depots, :type, :string
```

lpichler replied:
but when was this migration created the column was not here, so should I respect state of DB in that time or state of latest DB ?

jrafanie replied:
@lpichler When this migration was created, it would have been possible to run these migrations out of order if you were the author of the other migration.  Since we're long past this, I think it's fine to drop this as you have done... :+1:

jrafanie commented on db/migrate/20160203101130_add_orderable_to_orchestration_templates.rb:
```diff
@@ -1,5 +1,7 @@
 class AddOrderableToOrchestrationTemplates < ActiveRecord::Migration
-  class OrchestrationTemplate < ActiveRecord::Base; end
+  class OrchestrationTemplate < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
```
:+1: nice catch

jrafanie commented:
@lpichler @matthewd @Fryguy  I wonder if the migration specs could monkey patch/module prepend AR::Base's inheritance_column = to check if the table for the stub model already has a type column and warn about it if not (calling super regardless)
